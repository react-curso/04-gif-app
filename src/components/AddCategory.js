import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ categories }) => { 

     const [inputValue, setInputValue] = useState('');

     const handleInputChange = (e) => {
          setInputValue(e.target.value);
     }

     const handleSubmit = (e) => {
          e.preventDefault();
          if (inputValue.trim().length > 2) {
               categories(c => [ inputValue, ...c]);
               setInputValue('');

               
          }
     }

     return (
          <>
               <form onSubmit={handleSubmit} >
                    <input
                         type="text"
                         value={inputValue}
                         onChange={handleInputChange}
                    />
               </form>
          </>
     )
}

AddCategory.propTypes = {
     categories: PropTypes.func.isRequired
}

