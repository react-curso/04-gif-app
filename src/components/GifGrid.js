import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>

            { loading && <p>Loading</p> }            

            <div className="card-grid">
                {
                    images.map((img) => (

                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}
