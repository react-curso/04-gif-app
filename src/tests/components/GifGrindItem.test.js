import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

describe('pruebas de componente', () => {

    const parameters = {
        title: 'dad',
        url:'dadad'
    } 

    const wrapper = shallow(<GifGridItem {...parameters} />);

    test('Test Component', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('parrafo con title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( parameters.title );
        
    })

    test('test imagen ', () => {
        
        const img = wrapper.find('img').prop('src');
        const alt = wrapper.find('img').prop('alt');
        expect( img ).toBe( parameters.url );
        expect( alt ).toBe( parameters.title );

    })

    test('validar clase de div', () => {
        
        const div = wrapper.find('div');
        expect( div.hasClass('animate__fadeIn') ).toBe(true);

    })
    
})

