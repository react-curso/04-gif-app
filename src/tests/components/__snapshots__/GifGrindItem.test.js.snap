// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas de componente Test Component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="dad"
    src="dadad"
  />
  <p>
     
    dad
     
  </p>
</div>
`;
